src/main/java/io/cdap/plugin/dlp/DLPTransformPluginConfig.java

```diff
@@ -171,6 +186,26 @@ public void validate(FailureCollector collector, Schema inputSchema) {
               }
             }
           }
+
+          //Checking if location of wrapped key is same as DLP resource location
+          DlpTransformConfig transformProperties = config.getTransformProperties();
+          if (transformProperties instanceof CryptoDeterministicTransformationConfig) {
```

Comment:
this is an issue with the existing code, but there is way too much nesting in this method, makes it hard to read.

Can you change it to something like:

```
if (fieldToTransform == null) {
  return;
}
List<DlpFieldTransformationConfig> transformationConfigs;
try {
  transformationConfigs = parseTransformations();
} catch (Exception e) {
  collector.addFailure(...);
  return;
}
...
for (DlpFieldTransformationConfig config : transformationConfigs) {
  validateFieldConfig(config);
}
```

and move everything in the loop into a new validateFieldConfig() method